app/src/main/java/com/example/util/simpletimetracker/wear/WearService.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package com.example.util.simpletimetracker.wear
+
+import com.example.util.simpletimetracker.domain.interactor.PrefsInteractor
+import com.example.util.simpletimetracker.domain.interactor.RecordTagInteractor
+import com.example.util.simpletimetracker.domain.interactor.RecordTypeInteractor
+import com.example.util.simpletimetracker.domain.interactor.RunningRecordInteractor
+import com.example.util.simpletimetracker.domain.mapper.AppColorMapper
+import com.example.util.simpletimetracker.wearrpc.WearRPCServer
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
+import com.google.android.gms.wearable.WearableListenerService
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.runBlocking
+import javax.inject.Inject
+
+/**
+ * Landing point for messages coming to Mobile from Wear.
+ *
+ * Implemented as a Service so that it can receive messages even when the Mobile app is stopped or
+ * in the background.
+ */
+@AndroidEntryPoint
+class WearService : WearableListenerService() {
+
+    @Inject
+    lateinit var prefsInteractor: PrefsInteractor
+
+    @Inject
+    lateinit var recordTypeInteractor: RecordTypeInteractor
+
+    @Inject
+    lateinit var recordTagInteractor: RecordTagInteractor
+
+    @Inject
+    lateinit var runningRecordInteractor: RunningRecordInteractor
+
+    @Inject
+    lateinit var appColorMapper: AppColorMapper
+
+    override fun onRequest(nodeId: String, path: String, request: ByteArray): Task<ByteArray>? {
+        val rpc = WearRPCServer(
+            DomainAPI(
+                prefsInteractor,
+                recordTypeInteractor,
+                recordTagInteractor,
+                runningRecordInteractor,
+                appColorMapper,
+            ),
+        )
+        return runBlocking { Tasks.forResult(rpc.onRequest(path, request)) }
```

Comment:
I'll take a closer look at this one.

`runBlocking` was an easy option which enabled me to get the communication working in the first place. Since it runs in a `Service`, not a UI-rendering application, I wasn't as concerned by the block for the MVP.

Reply:
I'm unsure what I'm missing, but I am unable to remove the `runBlocking` call.

For example, I've tried switching to `async` (`launch` doesn't make sense because we need the return value)

```kotlin
val response = CoroutineScope(Dispatchers.Default).async { Tasks.forResult(rpc.onRequest(path, request)) }
return response.await()
```

But I get the following error message:
> Suspend function 'await' should be called only from a coroutine or another suspend function

The error makes sense. The `onRequest` method is not marked as a `suspend` function, nor can I mark it as such since it's an override. Therefore, the thread must block until the query is satisfied.

Reply:
Hmm, makes sense. I guess it can stay as it is for now. Need to do some debugging later, maybe WearableListenerService does indeed works on a background thread and blocking is ok here.